app/components/meeting/edit-meeting-modal.js

```diff
@@ -2,39 +2,60 @@ import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
 import { inject as service } from '@ember/service';
 import { action } from '@ember/object';
-import { task } from 'ember-concurrency';
+import { task, dropTask } from 'ember-concurrency';
+import { A } from '@ember/array';
+import moment from 'moment';
+import CONSTANTS from 'frontend-kaleidos/config/constants';
+import { fetchClosestMeetingAndAgendaId } from 'frontend-kaleidos/utils/meeting-utils';
 
 /**
  * @argument {meeting}
  * @argument {didSave}
  * @argument {onCancel}
  */
 export default class MeetingEditMeetingComponent extends Component {
+  @service store;
+  @service newsletterService;
   @service toaster;
 
+  @tracked isAnnexMeeting = false;
+  @tracked isEditingNumberRepresentation = false;
+
+  @tracked selectedMainMeeting;
+  @tracked isFinal;
   @tracked kind;
   @tracked startDate;
   @tracked extraInfo;
   @tracked _meetingNumber;
   @tracked _numberRepresentation;
 
+  currentYear = new Date().getFullYear();
+
   constructor() {
     super(...arguments);
 
+    const now = new Date();
+
     this.initializeKind.perform();
+    this.initializeMeetingNumber.perform();
+    this.initializeMainMeeting.perform();
 
-    this.meetingYear = this.args.meeting.plannedStart.getFullYear();
-    this.startDate = this.args.meeting.plannedStart;
+    this.meetingYear = this.args.meeting.plannedStart?.getFullYear() || this.currentYear;
+    this.startDate = this.args.meeting.plannedStart || new Date(now.getFullYear(), now.getMonth(), now.getDate(), 10, 0, 0);
     this.extraInfo = this.args.meeting.extraInfo;
-    this.meetingNumber = this.args.meeting.number;
+    this.isFinal = this.args.isFinal ?? false;
```

Comment:
`this.isFinal` seems not to be used. Argument `isFinal` is never passed to the component.

Reply:
Still confused about the `this.isFinal`. Since `this.args.isFinal` is always `undefined` (because not passed to the component) `this.isFinal` will always be `false`. The property is nowhere manipulated; only used to set `this.args.meeting.isFinal = this.isFinal` when saving the meeting. Is the tracked property needed? And does it need to be set to `false` on each save of the meeting? I don't see how it relates to the saved changes.

Behaviour I would expect is `isFinal` to be `false` on creation of a new meeting (that can be covered by adding the property on `createRecord('meeting', ...)` and leaving it untouched on edit of the meeting? Or am I missing something here @ValenberghsSven ?

Reply:
That line should be `this.isFinal = this.args.meeting.isFinal ?? false`, so it follows the expected behaviour, I just missed the `.meeting`